pkg/ebpf/bytecode/runtime/runtime_asset.go

```diff
@@ -30,6 +31,24 @@ var (
 	}
 )
 
+// RuntimeCompilationEnabled indicates whether or not runtime compilation is enabled
+var RuntimeCompilationEnabled = false
+
+// CompilationResult enumerates runtime compilation success & failure modes
+type CompilationResult int
+
+const (
+	notAttempted CompilationResult = iota
+	kernelVersionErr
+	verificationError
+	outputDirErr
+	outputFileErr
+	newCompilerErr
+	compilationErr
+	resultReadErr
+	compilationSuccess
```

Comment:
nit: It might be nice to have success be a value of `1`.  That way we know anything `> 1` is an error. It might make creating some of the graphs/dashboard easier, I think.

Reply:
My plan for how to translate this data to graphs/dashboards is to emit 2 metrics in the backend: `system_probe.<asset_name>.runtime_compilation.success` + `system_probe.<asset_name>.runtime_compilation.failure`, with a tag attached to the failure metric indicating the reason for failure. That way, we can easily get a count of total successes/failures & while also being able to see count of the failures by type. Thoughts?

Reply:
That sounds good. I do think my suggestion would make that translation logic clearer, especially if we add more error conditions to this enum, say from kernel header downloading.

Reply:
This is ultimately a super minor thing, so do as you please. 😁

Reply:
It's an easy change to make, I don't mind :)